src/utilities/use-deep-compare-ref.tsx

```diff
@@ -4,6 +4,18 @@ import isEqual from 'lodash/isEqual';
 type DependencyList = ReadonlyArray<unknown>;
 type Comparator = (a: DependencyList, b: DependencyList) => boolean;
 
+/**
+ * useDeepCompare is valuable to keep a consistent dependency list across reference changes
+ * @param dependencies A dependency array like React.useEffect/React.useCallback/React.useMemo
+ * @param comparator A optional function to compare dependencies that'll default to a deep comparison
+ * @returns A dependency list
```

Comment:
```suggestion
 * @returns A dependency array
```

Reply:
Going keep it as a list for the moment since it's how react reference to it.